ambry-cloud/src/main/java/com/github/ambry/cloud/azure/AzureStorageContainerMetricsCollector.java

```diff
@@ -0,0 +1,96 @@
+/**
+ * Copyright 2024 LinkedIn Corp. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ */
+package com.github.ambry.cloud.azure;
+
+import com.codahale.metrics.MetricRegistry;
+import com.github.ambry.config.VerifiableProperties;
+import com.github.ambry.utils.Utils;
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.concurrent.ScheduledExecutorService;
+import java.util.concurrent.TimeUnit;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+
+/**
+ * A class that aggregates container metrics. We do _NOT_ want to emit per-container metrics because then the number
+ * of metrics increases proportionally overwhelming telemetry. This was attempted and failed.
+ *
+ * A daemon will run at regular intervals emitting aggregate metrics in a controlled and predictable manner.
+ * This is a singleton class to avoid multiple collector threads.
+ */
+public class AzureStorageContainerMetricsCollector {
+  private final AzureMetrics metrics;
+  private final ConcurrentHashMap<Long, AzureStorageContainerMetrics> metricMap;
```

Comment:
we actually don't need to create a new `AzureStorageContainerMetrics` class, we can just put an AtomicLong here in the map.

Reply:
Right, but I imagine there could be more metrics in the future and I am gently guiding the future dev to add metrics here rather than creating more maps. Is that good ?